src/markdown/parser.cr

```diff
@@ -452,6 +463,36 @@ class Markdown::Parser
     String.new(Slice.new(str + bracket_idx + 2, paren_idx - 1))
   end
 
+  # Checks if the following sequence of characters is a checkbox.
+  # A valid checkbox is made up of 3 characters and looks like
+  # this `[ ]` or this `[x]` with the latter being a checked one.
+  def check_checkbox(str, pos, bytesize)
```

Comment:
Isn't this method a bit too complex? Why not simply
```cr
def check_checkbox(str, pos, bytesize)
  return false unless pos + 3 < bytesize

  return false unless str[pos] == '['.ord
  return false unless str[pos + 1] == ' '.ord || str[pos + 1] == 'x'.ord
  return false unless str[pos + 2] == ']'.ord

  true
end
```

Is it required to be whitespace-insensitive? Is `[     ]` really a checkbox?